examples/cross-lingual-transfer/main.py

```diff
@@ -0,0 +1,168 @@
+import os
+import torch
+import numpy as np
+from pathlib import Path
+
+import evaluate
+from transformers import TrainingArguments, Trainer
+from transformers import AutoModelForSequenceClassification, AutoTokenizer
+from datasets import load_dataset
+
+import mlflow
+from transformers.integrations import MLflowCallback
+from aim.hugging_face import AimCallback
+
+logging.basicConfig(
+    format="%(asctime)s.%(msecs)03d %(levelname)s %(module)s->%(funcName)s: %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+    level=logging.DEBUG,
+)
+
+
+class Solver:
+    def __init__(self, args) -> None:
+        self.args = args
+        self.args.output_dir = Path(self.args.output_dir)
+
+        self.experiment_name = f"xnli_{self.args.technique}"
+        self.device = "cuda" if torch.cuda.is_available() else "cpu"
+
+    def compute_metrics(self, eval_pred):
+        logits, labels = eval_pred
+        predictions = np.argmax(logits, axis=-1)
+        return self.metric.compute(predictions=predictions, references=labels)
+
+    def tokenize_function(self, examples):
+        return self.tokenizer(
+            examples["premise"],
+            examples["hypothesis"],
+            padding="max_length",
+            truncation=True,
+        )
+
+    def run(self):
+        for model_name in self.args.model_names:
+            logging.info(f"solving XNLI using {model_name}")
+            # dataset
+
+            self.tokenizer = AutoTokenizer.from_pretrained(model_name)
+
+            train_dataset = (
+                load_dataset("xnli", "en").shuffle(seed=self.args.seed)["train"]
+            ).select(range(self.args.num_train_batches))
+            train_dataset_tokenized = train_dataset.map(
+                self.tokenize_function, batched=True
+            )
+
+            eval_datasets = {}
+            for eval_dataset_name in self.args.eval_datasets_names:
+                eval_dataset = load_dataset("xnli", eval_dataset_name).shuffle(
+                    seed=self.args.seed
+                )["validation"]
+                eval_datasets[eval_dataset_name] = eval_dataset.map(
+                    self.tokenize_function, batched=True
+                )
+
+            # model
+            model = AutoModelForSequenceClassification.from_pretrained(
+                model_name, num_labels=train_dataset.features["label"].num_classes
+            ).to(self.device)
+
+            if self.args.technique == "feature-extraction":
+                for name, param in model.named_parameters():
+                    # choose whatever you like here
+                    if not name.startswith("classifier"):
+                        param.requires_grad = False
+
+            # metric
+            self.metric = evaluate.load("accuracy")
+
+            # trackers
+            mlflow.set_tracking_uri(self.args.output_dir.joinpath("mlruns"))
+            os.environ["MLFLOW_EXPERIMENT_NAME"] = self.experiment_name
+
+            aim_callback = AimCallback(
```

Comment:
Do we need to add `AimCallback`? Or the data should be synced with Aim using `aimlflow` CLI?

Reply:
Nice question @alberttorosyan.
In fact MLflow's callback is creating solid names (e.g. `eval_de_loss`). Thus making it hard(if not impossible) to explore the metrics on Aim UI, so using aimlflow is not the best option for MLflow's huggingface transformers callback.
However Aim's built-in callback for huggingface transformers has much better functionality and adds context to each metric (e.g. `loss` `context={"sub_dataset": "de", "subset": "eval"}`), making it super easy to work with these metrics on UI. I have consider these two cases in my use-case/blog post.